projects/core/src/product/facade/product-search.service.ts

```diff
@@ -32,10 +32,15 @@ export class ProductSearchService {
     );
   }
 
-  getSearchResults(): Observable<ProductSearchPage> {
+  getSearchResults(query?: string): Observable<ProductSearchPage> {
```

Comment:
this `getSearchResults` is only called in `product-list.component.ts`. So, I think, we can just remove the `filter` and move the `tap` logic to component. Because, the `SearchConfig` also required when call `search`.